src/commons/components/ceph-dashboard/ceph-dashboard.js

```diff
@@ -0,0 +1,269 @@
+
+  app.component("cephDashboard", {
+    bindings: {
+      cephCluster: "<",
+      chartData: "=?",
+      showClusterData: "=?",
+      showSystemPerformance: "=?"
+    },
+    controller: "cephDashboardController",
+    controllerAs: "vm",
+    templateUrl: "/commons/components/ceph-dashboard/ceph-dashboard.html"
+  });
+
+  function cephDashboardController($scope, $rootScope, dashboardStore){
+    var vm = this,
+    i,
+    checkTime,
+    rawStorageUtilizationXData = ["percent"],
+    rawStorageUtilizationYData = ["used"];
+
+    vm.$onChanges = function(changesObj){
+        if(vm.showClusterData){
+            clusterData(vm.cephCluster);
+        }
+        clusterHost(vm.cephCluster);
+        clusterMon();
+        clusterOsd();
+        vm.poolBarChartTitleData = poolChartTitleData();
+        vm.poolBarChartData = poolBarChartData();
+        vm.rbdBarChartTitleData = rbdChartTitleData();
+        vm.rbdBarChartData = rbdBarChartData();
+        rawStorageUtilization(vm.cephCluster);
+        if(vm.showSystemPerformance && vm.cephCluster){
+            defaultSystemPerformanceSettings(vm.cephCluster.node_summaries);
+            vm.heatMapData = dashboardStore.systemPerformance(vm.cephCluster.node_summaries);
+        }
+    };
+
+    function clusterData(clusterData){
+        var clusterCountData = clusterData.cluster_count,
+        downClusters = clusterCountData.status.total -
+                    (clusterCountData.status.HEALTH_OK ? clusterCountData.status.HEALTH_OK : 0 +
+                    clusterCountData.status.HEALTH_WARN ? clusterCountData.status.HEALTH_WARN : 0 +
+                    clusterCountData.status.healthy ? clusterCountData.status.healthy : 0);
+
+        vm.clusterStatus = {
+           "title":"Cluster",
+           "count": clusterCountData.status.total ? clusterCountData.status.total : 0,
+           "notifications":[
+            {
+                "iconClass":"fa fa-arrow-circle-o-down",
+                "count": downClusters,
+            },
+            {
+                "iconClass":"pficon  pficon-resources-almost-full",
+                "count": (clusterCountData.near_full ? clusterCountData.near_full : 0),
+            },
+            {
+                "iconClass":"pficon pficon-error-circle-o",
+                "count":(clusterCountData.critical_alerts ? clusterCountData.critical_alerts : 0),
+            },
+            {
+                "iconClass":"pficon pficon-warning-triangle-o",
+                "count":(clusterCountData.warning_alerts ? clusterCountData.warning_alerts  : 0),
+            },
+            {
+                "iconClass":"pficon pficon-flag",
+                "count": 0,
+                "text": " Cluster have host quorum"
+            }]
+        };
+    }
+
+    function clusterHost(clusterData){
+        var hostData = clusterData.hosts_count;
+        vm.hostStatus = {
+        "title":"Hosts",
+        "count": hostData.total,
+        "notifications":[
+        ]};
```

Comment:
@cloudbehl It can be written as "notifications": [] instead of closing "]" on next line.